src/commands/edge_app.rs

```diff
@@ -234,6 +236,38 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn run(&self, path: &Path, secrets: Vec<(String, String)>) -> Result<(), anyhow::Error> {
+        let address_shared = Arc::new(Mutex::new(None));
+        let address_clone = address_shared.clone();
+
+        let runtime = tokio::runtime::Runtime::new().unwrap();
+        let path = path.to_path_buf();
+        runtime.block_on(async {
+            tokio::spawn(async move {
+                let address = run_server(path.as_path(), secrets).await.unwrap();
+                let mut locked_address = address_clone.lock().unwrap();
```

Comment:
Hm, strange - it is used here and immediately reassigned on next line... Will lock persist after reassignment?

Reply:
I am not sure I understand

Reply:
First line is
locked_address = address_clone.....
second is
*locked_address = Some(address)

My question is if first line actually do smth as you imediately replace locked_address value.

Reply:
I still don't understand... it is mutex that calls lock(). I lock first then change the value

Reply:
Ok, just was trying to figure out how it works - nvm